lib/elixir/lib/io/ansi.ex

```diff
@@ -171,27 +171,30 @@ defmodule IO.ANSI do
   defsequence(:home, "", "H")
 
   @doc """
-  Sends cursor to the absolute position specified by `lines`, `columns`, 
-  with 0,0 being the top left corner.
+  Sends cursor to the absolute position specified by `line` and `column`.
+
+  Line `0` and column `0` would mean the top left corner.
   """
   @spec cursor(integer, integer) :: String.t()
-  def cursor(line, column) when line >= 0 and column >= 0, do: "\e[#{line};#{column}H"
+  def cursor(line, column)
+      when is_integer(line) and line >= 0 and is_integer(column) and column >= 0,
```

Comment:
@whatyouhide this is how `mix format` formatted it but it looks a little ugly. Think I should make a `is_positive_integer` guard / do we have one somewhere?

Reply:
We don't but this formatting is fine. When `when` gets split on its own line, we just use the `do`/`end` syntax:

```elixir
def cursor(line, column) 
    when is_integer(line) and line >= 0 and is_integer(column) and column >= 0 do
  ...
end
```